src/models/continuity/ContGateTurn.ts

```diff
@@ -30,17 +32,56 @@ export default class ContGateTurn
 
   getHumanReadableText(): string {
     if (this.humanReadableText !== "") return this.humanReadableText;
-    // TODO: Implement
-    return "";
+    const rotation: string = Math.sign(this.angle) === 1 ? "CW" : "CCW";
+    return `GT${this.marchType} ${this.duration} COUNTS ${Math.abs(
+      this.angle
+    )} DEGREES ${rotation} ABOUT POINT [${this.centerPoint}]`;
```

Comment:
Are we expecting to make a new dot type for every gate turn group? I wonder if we can design this so that it can accommodate multiple gate turn groups and centers?

Let's leave an issue to track it if we aren't going to get to it here.

Reply:
Yeah actually I was a bit confused as to how multiple centers would work. Would dots within the gateturn continuity rotate about the center closest to them? And if so, what would happen if while rotating a dot becomes closer to a new center? I've never done a gate turn in current Calchart so I'm not sure how this is supposed to work... I currently just made it for a single center.

Reply:
I'll mark down here how classic CalChart achieves this. The way CalChart does this is pretty powerful, but a bit advanced.

Let's say we start with a field that looks like this. We have two clusters of filled dots that we want to engage in gate turns, but they have the same type. We want [0,1,2,,43] to rotate 90 degrees clockwise around 0, and we want [5,6,7,8,9] to rotate 90 degrees clockwise around 5.

<img width="1312" alt="Screen Shot 2021-01-02 at 4 56 45 PM" src="https://user-images.githubusercontent.com/3744644/103469587-8f62ba80-4d1b-11eb-8eb0-e42a65c0bc85.png">

We can start by going to the continuity editor, and adding a 'ROTATE' continuity for the filled dot type. We will need to fill in 3 arguments to our continuity -- the help documentation tells us what those should be:

<img width="812" alt="Screen Shot 2021-01-02 at 4 42 37 PM" src="https://user-images.githubusercontent.com/3744644/103469436-a4d6e500-4d19-11eb-82ac-c2ff6569cbef.png">

I am going to use the following command to say that I want to rotate -90 degrees (that is, 90 degrees clockwise) over 4 counts, around 'r1' (this is called 'ref point 1; I'll go a little more into this in a moment). Then I'll add 'mtrm e' to say 'mark time facing east for any remaining counts'. The final continuity looks like this:

```
rotate -90 4 r1
mtrm e
```

The key to this is 'r1' ('ref point 1'), which is serving as the 'pivot point' argument. In classic CalChart, every dot on the field has 3 invisible 'ref points' associated with it. These ref points are invisible by default, but you can use the 'reference group' dropdown to make them visible. Once visible, you can move the ref points around in the same way that you would move around any other point on the field.

<img width="585" alt="Screen Shot 2021-01-02 at 5 01 25 PM" src="https://user-images.githubusercontent.com/3744644/103469629-2af42b00-4d1c-11eb-8e7c-965c0306b6e7.png">

Every dot has an independent R1, and every dot has been instructed to rotate around their R1, so you could theoretically give them all unique pivot points. In this case, we don't want to do that -- we want the pivot point for the first five dots to be under point 0, and we want the pivot point for the second five dots to be under point 5. By default, the R1 for each dot is going to be positioned right under their respective dots. So first, I select the ref points that appear in the positions of the first five dots:

<img width="221" alt="Screen Shot 2021-01-02 at 5 04 19 PM" src="https://user-images.githubusercontent.com/3744644/103469666-92aa7600-4d1c-11eb-8073-f8a20ed11b9a.png">

And then I move all of those R1 points underneath point 0. The result looks like this; note that the ref points appear white, and the real 'marcher' points appear in purple. The purple dots outline our formation; the white ones are our 'pivot points'.

<img width="176" alt="Screen Shot 2021-01-02 at 5 05 42 PM" src="https://user-images.githubusercontent.com/3744644/103469683-c5546e80-4d1c-11eb-8384-8a476d446204.png">

We can likewise move the white 'pivot points' for the second 5 dots underneath point 5. The result looks like this:

<img width="518" alt="Screen Shot 2021-01-02 at 5 06 47 PM" src="https://user-images.githubusercontent.com/3744644/103469701-eae17800-4d1c-11eb-85eb-918ad75bead7.png">

All of the dots have the same 'dot type' (they are all filled dots, as opposed to X dots or slash dots), but they do not all rotate around the same pivot point.

Reply:
^ This isn't to say this is how it should be done in the app. I think if there is a simpler way to achieve something like this, that would be great. Maybe the answer is just 'one pivot point per dot type'; but I could see the number of dot types exploding in some cases so that may not be so graceful.

Reply:
Thanks a bunch @kmdurand , that was incredibly helpful! I agree that the number of dot types could get messy, I'm thinking back to SS14 of the pride show http://calband.github.io/calchart-viewer/?show=the-pride-show, where everyone in the band is doing a bunch of gate turns and it gets crazy.
<img width="599" alt="image" src="https://user-images.githubusercontent.com/29964426/103471461-9dbdd000-4d35-11eb-9ae1-b5cd48d2f8ce.png">
I'm interested to see how we could implement this centering format in the new calchart... Maybe once @MichaelTamaki  gets his dot ID PR in we can use the centerPoints array to map from ID to center.

Reply:
Thank you for the detailed explanation @kmdurand! Let's make sure to capture this in an issue if we're going to implement this in a different PR.